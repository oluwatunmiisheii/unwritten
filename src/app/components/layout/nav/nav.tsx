'use client'

import React from 'react'

export const Nav = () => {
  function formatDate(date = new Date()) {
    const weekday = date.toLocaleDateString('en-GB', { weekday: 'long' })
    const month = date.toLocaleDateString('en-GB', { month: 'short' })
    const year = date.getFullYear()

    return `${weekday}, ${month} ${year}`
  }

  return (
    <header>
      <nav className="py-20 px-8 md:px-0 max-w-5xl mx-auto">
        <ul className="flex flex-col sm:flex-row justify-between items-center">
          <li>
            <h5 className="font-pacifico font-bold text-2xl">PhotoSearch.</h5>
          </li>
          <li>
            <p>{formatDate()}</p>
          </li>
        </ul>
      </nav>
    </header>
  )
}
