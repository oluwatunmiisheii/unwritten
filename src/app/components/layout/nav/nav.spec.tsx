import { Nav } from './nav'
import { expect, test, describe, vi } from 'vitest'
import { render } from '@testing-library/react'

describe('Nav', () => {
  test('should render correctly', () => {
    const fixedDate = new Date('2024-01-15')
    vi.spyOn(global, 'Date').mockImplementation(() => fixedDate)
    const { container } = render(<Nav />)
    expect(container).toMatchSnapshot()

    vi.restoreAllMocks()
  })
})
