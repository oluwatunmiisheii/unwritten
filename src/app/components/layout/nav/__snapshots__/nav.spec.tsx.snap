// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Nav > should render correctly 1`] = `
<div>
  <header>
    <nav
      class="py-20 px-8 md:px-0 max-w-5xl mx-auto"
    >
      <ul
        class="flex flex-col sm:flex-row justify-between items-center"
      >
        <li>
          <h5
            class="font-pacifico font-bold text-2xl"
          >
            PhotoSearch.
          </h5>
        </li>
        <li>
          <p>
            Monday, Jan 2024
          </p>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
